ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java

```diff
@@ -225,31 +274,219 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
                 .toLocalDateTime();
     }
 
+
     /**
      * This method validates that each data field necessary has the required and correct data type
+     * @return false if any test is invalid, else true
      */
-    private void validateData() {
+    private boolean validateData() {
+        // The final outcome of the validation
+        boolean result = true;
+
+        // Validation outcome for class number
+        boolean testDaysOfTheWeek = true;
+        testDaysOfTheWeek = validateDates();
 
+        // Validation outcome for department
+        boolean testDepartmentName = true;
+        testDepartmentName = validateDepartment();
+
+        // Validation outcome for class number
+        boolean testClassNumber = true;
+        testClassNumber = validateClassNumber();
+
+        // Validation outcome for class name
+        boolean testClassName = true;
+        testClassName = validateClassName();
+
+        // Validation outcome for the start and end time
+        boolean testTime = true;
+        testTime = validateTime();
+
+        // Check all final validation outcome, if any result to false, then set final result to false
+        if(testDaysOfTheWeek == false || testDepartmentName == false || testClassNumber == false || testClassName == false || validateTime() == false) {
```

Comment:
Don't necessarily need to do it like this. You could just do

return (testDaysOfTheWeek == false || testDepartmentName == false || testClassNumber == false || testClassName == false || validateTime() == false) ;

If you wanted it even shorter you could just do:

return validateDates() && validateDepartment() && validateClassNumber() && validateClassName() && validateTime()

Reply:
Sounds good! Thank you!